src/page-container/MainPageContainer.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+
+import MainContainer from '../main-container';
+import LayoutActionsContext from '../shared/LayoutActionsContext';
+import NavigationItemContext from '../navigation-item/NavigationItemContext';
+import deferExecution from '../utils/defer-execution';
+
+import PageContainerContext from './PageContainerContext';
+import usePageManager from './usePageManager';
+
+import styles from './PageContainer.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The components to render within the context of the MainPageContainer. Any
+   * Pages rendered as children will be managed by the MainPageContainer.
+   */
+  children: PropTypes.node,
+};
+
+/**
+ * The MainPageContainer defines a content region within the application
+ * where Pages can be rendered. Any Page components rendered within a
+ * MainPageContainer will be portaled into the MainPageContainer's 'main'
+ * element.
+ *
+ * Because the MainPageContainer renders a 'main' element, only a single instance
+ * should be rendered and visible at any given moment. The `renderPage` prop
+ * provided by layouts and navigation items should be used whenever possible;
+ * direct usage of the MainPageContainer is not recommended.
+ */
+const MainPageContainer = ({
+  children,
+}) => {
+  const navigationItem = React.useContext(NavigationItemContext);
+  const layoutActions = React.useContext(LayoutActionsContext) || {};
```

Comment:
What are the scenarios where this evaluates to false, when the provider isn't defined?

Reply:
I was trying to avoid setting a default value for that context because I was doing some explicit overrides in future work. But I took another look at things and I don't think those overrides will be necessary any more. So I removed this and added a default value.